models/issues/issue.go

```diff
@@ -1251,6 +1251,8 @@ func (opts *IssuesOptions) setupSessionNoLimit(sess *xorm.Session) {
 	if opts.ProjectID > 0 {
 		sess.Join("INNER", "project_issue", "issue.id = project_issue.issue_id").
 			And("project_issue.project_id=?", opts.ProjectID)
+	} else if opts.ProjectID == -1 { // -1 means not in any project
```

Comment:
```suggestion
	} else if opts.ProjectID == db.NoneID { // -1 means not in any project
```
also means this PR should be blocked until #22423 has been merged

Reply:
or we extract that part from #22423 and incorporate it here already. I'm also fine with that.

Reply:
I'd still like adding the constant in this PR.

Reply:
I think I have done that.

Reply:
~~According to the diff: Apparently not?
Did you perhaps not push your changes?~~

Reply:
Ah, no. GitHub is a bad software which simply didn't show me that it only used an outdated diff.